Foundation/NSCFBoolean.swift

```diff
@@ -78,7 +78,10 @@ internal class __NSCFBoolean : NSNumber {
     override var objCType: UnsafePointer<Int8> {
         // This must never be fixed to be "B", although that would
         // cause correct old-style archiving when this is unarchived.
-        return UnsafePointer<Int8>(bitPattern: UInt(_NSSimpleObjCType.Char.rawValue.value))!
+        func _objCType(_ staticString: StaticString) -> UnsafePointer<Int8> {
```

Comment:
I discarded following codes that rewrote `_NSSimpleObjCType` in `NSObjCRuntime.swift`, since memory layout is not compatible with current implementation.
https://gist.github.com/norio-nomura/a1383f258c38bb1e081597a538f46e8f

Should we create another static values for `@encode`?
I feel we should do re-writing `_NSSimpleObjCType` instead of adding them, and doing that with this PR will make it too large.